pkg/sql/logictest/testdata/logic_test/statement_statistics

```diff
@@ -97,6 +97,11 @@ SELECT x FROM test WHERE y = 1/z
 statement ok
 SET CLUSTER SETTING debug.panic_on_failed_assertions = true; RESET CLUSTER SETTING debug.panic_on_failed_assertions
 
+statement ok
+SHOW application_name
```

Comment:
what happens if I `SHOW some_unrecognized_var`?

Reply:
Then the thing will log `SHOW some_unrecognized_var`. Is that not what we want? This would help us determine when clients want to enquire about pg config variables/parameters that we don't support yet.

Reply:
Hm. That sounds useful... but i’m a little uncomfortable with reporting an arbitrary (potentially user-entered) string. I don’t have more of an answer than that yet though.

Reply:
The "problem" exists in the same way for SHOW CLUSTER SETTING (which we report unchanged). 

It also exists for SET (all forms).

It also exists for the KV options names of the CCL statements.

What do you think? Should we scrub all those too?

Reply:
I'm still scratching my chin over here and I don't have what feels like a good answer yet.

Operating under the assumption that var names probably come from some off-the-shelf ORM setting options or something, I agree we'd get significant value in seeing what, including unsupported, things they try to set. Also, it seems like they're probably not user-speicifc.


Our options appear to be:
1) Just report var names as they appear (i.e. merge this). Since they're probably not user-speicifc, this is probably fine.
2) Only report recognized var names. This would be difficult -- we'd need the statement-specific logic that handles vars in the scrubbing, and gets messy fast. Also, we no longer have visibility into unsupported var attempts.
3) Make a generic string scrubber backed by a big, generated whitelist, containing all known settings names, all known var names, kv option names, and maybe we could dig up the list of all valid pg var names to add to it as well and capture some unsupported attempts too. This could avoid the messy dependencies in the scrubber, but maintaining this list (either by hand or via tooling) adds significant cost to this approach.

Anyway, I like I said, I still don't feel like I have a Good Answer.

Reply:
I appreciate the discussion however we must take an immediate next step:

- either you think this topic is indeed a serious consideration, and in that case I must immediately merge (with a backport to 2.0) a PR that ensures the cluster settings and the parameter to SET are scrubbed in all cases, because your point here is that we should be careful about user mistakenly embedding PII in their variable names;

- or this is merely a discussion to be had, in which case we can probably merge this as-is and then table a discussion about how to do better between now and october.

Which one is it? If it's the first, I can look at it today.